src/providers/osu.ts

```diff
@@ -0,0 +1,34 @@
+import { OAuthConfig, OAuthUserConfig } from "."
+
+export interface OsuProfile {
+  id: string
+  username: string
+  avatar_url: string
+}
+
+export default function Osu<P extends Record<string, any> = OsuProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "osu",
+    name: "Osu!",
+    type: "oauth",
+    token: "https://osu.ppy.sh/oauth/token",
+    authorization: {
+      url: "https://osu.ppy.sh/oauth/authorize",
+      params: {
+        scope: "identify"
+      }
+    },
+    userinfo: "https://osu.ppy.sh/api/v2/me",
+    profile(profile) {
+      return {
+        id: profile.id,
+        email: null,
```

Comment:
So this Provider does not return `email`, huh? :confused: Do you know if there is any way of retrieving one? If not, then it is fine, I just wanted to run a question round on this.

Reply:
To my knowledge, no. 
Since you can't even see your own current Email in Osu! without asking the support I wasn't able to fine anything in the documentation.

Reply:
Acknowledged. Thank you for explaining.